src/main/kotlin/org/opensearch/replication/ReplicationPlugin.kt

```diff
@@ -312,19 +327,24 @@ internal class ReplicationPlugin : Plugin(), ActionPlugin, PersistentTaskPlugin,
                 REPLICATION_LEADER_THREADPOOL_QUEUE_SIZE, REPLICATION_FOLLOWER_CONCURRENT_READERS_PER_SHARD,
                 REPLICATION_FOLLOWER_RECOVERY_CHUNK_SIZE, REPLICATION_FOLLOWER_RECOVERY_PARALLEL_CHUNKS,
                 REPLICATION_PARALLEL_READ_POLL_INTERVAL, REPLICATION_AUTOFOLLOW_REMOTE_INDICES_POLL_INTERVAL,
-                REPLICATION_AUTOFOLLOW_REMOTE_INDICES_RETRY_POLL_INTERVAL, REPLICATION_METADATA_SYNC_INTERVAL)
+                REPLICATION_AUTOFOLLOW_REMOTE_INDICES_RETRY_POLL_INTERVAL, REPLICATION_METADATA_SYNC_INTERVAL,
+                REPLICATION_PLUGIN_ENABLED)
     }
 
     override fun getInternalRepositories(env: Environment, namedXContentRegistry: NamedXContentRegistry,
                                          clusterService: ClusterService, recoverySettings: RecoverySettings): Map<String, Repository.Factory> {
-        val repoFactory = Repository.Factory { repoMetadata: RepositoryMetadata ->
-            RemoteClusterRepository(repoMetadata, client, clusterService, recoverySettings, replicationMetadataManager, replicationSettings) }
-        return mapOf(REMOTE_REPOSITORY_TYPE to repoFactory)
+        return if (isPluginDisabled) Collections.emptyMap()
+        else {
+            val repoFactory = Repository.Factory { repoMetadata: RepositoryMetadata ->
+                RemoteClusterRepository(repoMetadata, client, clusterService, recoverySettings, replicationMetadataManager, replicationSettings)
+            }
+            mapOf(REMOTE_REPOSITORY_TYPE to repoFactory)
```

Comment:
we will have to clean-up internal repositories as well based on this update: https://github.com/opensearch-project/cross-cluster-replication/blob/main/src/main/kotlin/org/opensearch/replication/repository/RemoteClusterRepositoriesService.kt#L37

Reply:
Intention to use this setting isn't to disable the ongoing replication but rather a enabling mechanism for control the launch.